docs/fsharp/style-guide/formatting.md

```diff
@@ -0,0 +1,506 @@
+
+```fsharp
+// A is a top-level module.
+module A
+
+let function1 a b = a - b * b
+```
+
+```fsharp
+// A1 and A2 are local modules.
+module A1 =
+    let function1 a b = a*a + b*b
+
+module A2 =
+    let function2 a b = a*a - b*b
+```
+
+### Formatting object expressions and interfaces
+
+Object expressions and interfaces should be aligned in the same way with `member` being indented after 4 spaces.
+
+```fsharp
+let comparer =
+    { new IComparer<string> with
+          member x.Compare(s1, s2) =
+              let rev (s : String) =
+                  new String (Array.rev (s.ToCharArray()))
+              let reversed = rev s1 i
+              reversed.CompareTo (rev s2) }
+```
+
+### Formatting whitespace in expressions
+
+Avoid extraneous whitespace in F# expressions.
+
+```fsharp
+// OK
+spam (ham.[1])
+
+// Not OK
+spam ( ham.[ 1 ] )
+```
+
+Named arguments should also not have space surrounding the `=`:
+
+```fsharp
+// OK
+let makeStreamReader x = new System.IO.StreamReader(path=x)
+
+// Not OK
+let makeStreamReader x = new System.IO.StreamReader(path = x)
+```
+
+## Formatting blank lines
+
+* Separate top-level function and class definitions with two blank lines.
+* Method definitions inside a class are separated by a single blank line.
+* Extra blank lines may be used (sparingly) to separate groups of related functions. Blank lines may be omitted between a bunch of related one-liners (e.g. a set of dummy implementations).
+* Use blank lines in functions, sparingly, to indicate logical sections.
+
+## Formatting comments
+
+Generally prefer multiple double-slash comments over ML-style block comments.
+
+```fsharp
+// Prefer this style of comments when you want
+// to express written ideas on multiple lines.
+
+(*
+    Generally avoid these kinds of comments.
+*)
+```
+
+Start inline comments with a new line, and captialize the first letter.
+
+```fsharp
+let f x = x + 1 // Increment by one.
+```
+
+## Naming conventions
+
+### Use camelCase for class-bound, expression-bound and pattern-bound values and functions
+
+It is common and accepted F# style to use camelCase for all names bound as local variables or in pattern matches and function definitions.
+
+```fsharp
+// OK
+let addIAndJ i j = i + j
+
+// Bad
+let addIAndJ I J = I+J
+
+// Bad
+let AddIAndJ i j = i + j
+```
+
+Locally-bound functions in classes should also use camelCase.
+
+```fsharp
+type MyClass() =
+
+    let doSomething () =
+
+    let firstResult = ...
+
+    let secondResult = ...
+
+    member x.Result = doSomething()
+```
+
+### Use camelCase for internal and private module-bound values and functions
+
+Use camelCase for private module-bound values, including the following:
+
+* Ad hoc functions in scripts
+
+* Values making up the internal implementation of a module or type
+
+```fsharp
+let emailMyBossTheLatestResults =
+    ...
+```
+
+In large assemblies and implementation files, PascalCase is sometimes used to indicate major internal routines.
```

Comment:
I don't think there's a need to promulgate this as the value of such a scheme is minimal, versus the risk of people using this to further arbitrary inconsistency. There are many better ways to achieve similar demarcation. Next we'll be advocating #regions !

Reply:
Deaded it